service/prometheus/update.go

```diff
@@ -31,20 +31,6 @@ func UpdateConfig(promcfg config.Config, scrapeConfigs []config.ScrapeConfig) (c
 // isManaged returns true if the given scrape config is managed by the prometheus-config-controller,
 // false otherwise.
 func isManaged(scrapeConfig config.ScrapeConfig) bool {
-	// TODO: the static config detection can be removed once all installations use k8s sd.
-	for _, targetGroup := range scrapeConfig.ServiceDiscoveryConfig.StaticConfigs {
-		if _, ok := targetGroup.Labels[ClusterLabel]; ok {
-			return true
```

Comment:
This is the older static config support. No installations have static configuration anymore, so can be removed.